### ISSUE TITLE
Allow running chains in parallel 

### ISSUE THREAD

**Issue by jgabry** [stan-dev/cmdstanr#3]
Should we use something like the parallel R package (or one of the many others), or do parallelization like in the CmdStan manual through the command line?

**Comment by jgabry** [comment]
We can easily start multiple processes at the same time by running `processx::process$new(...)` once for each chain, so that’s one possible way do to parallelization.

But I haven’t yet figured out a user friendly way to report the progress of the chains when they are running simultaneously (related to #18)

**Comment by jgabry** [comment]
As I just commented in #18, one possible approach would be to always (or optionally) run parallel chains by starting multiple background processes. This way the user's R session is free and we can provide a way for them to check the status of the processes. For example, maybe to the user it could look something like this: 

```r
fit <- mod $sample(num_chains = 4) 
fit$get_status() # get report on all chains if no argument specified 
fit$get_status(chain = 1) # check just chain 1
```

I don't know if this is the best approach or what the method for checking the progress should be called, but just wanted to throw this out there as an idea. I haven't tried it before but if we do want to do this then it looks like we can start a background process using `processx::process$new(...)` and then presumably use the methods described here

https://processx.r-lib.org/#managing-external-processes

to check the status and whatever else we need to do.

**Comment by jgabry** [comment]
Closing this now that we have #61. Will open a separate issue for whether or not we want to allow the option of not having it be a blocking function.
